src/region.js

```diff
@@ -202,20 +202,29 @@ Marionette.Region = Marionette.Object.extend({
 
   // Destroy the current view, if there is one. If there is no
   // current view, it does nothing and returns immediately.
-  empty: function() {
+  empty: function(options) {
     var view = this.currentView;
+    var emptyOptions = options || {};
 
+    var preventDestroy = !!emptyOptions.preventDestroy;
     // If there is no view in the region
     // we should not remove anything
     if (!view) { return; }
 
     view.off('destroy', this.empty, this);
     this.triggerMethod('before:empty', view);
-    this._destroyView();
+    if (!preventDestroy) {
+      this._destroyView();
+    }
     this.triggerMethod('empty', view);
 
     // Remove region pointer to the currentView
     delete this.currentView;
+
+    if (preventDestroy) {
```

Comment:
I would just do `Marionette._getValue(options, 'preventDestroy', this)` if we want to support function prevent destroy otherwise `_.result(options, 'preventDestroy')`.

Otherwise +1